File: kotlin/observer/NewsLetter/observer/Subscriber.kt
package observer

import subject.Newsletter

class Subscriber(newsletter: Newsletter) : Observer {

    private var name: String? = null

    private var newsletter = Newsletter()

    init {
        this.newsletter = newsletter
        newsletter.registerObserver(this)
    }

    override fun update(name: String) {
        this.name = name
        println("New Subscriber: +$name added to the newsletter\n")
    }

}

File: python/observer/NewsLetter/observer/Subscriber.py
from observer.Observer import Observer
from subject.Newsletter import Newsletter


class Subscriber(Observer):

    def __init__(self, newsletter):
        self.__name = ''
        self.__newsletter = Newsletter()
        self.__newsletter = newsletter
        self.__newsletter.register_observer(self)

    def update(self, name):
        self.__name = name
        print("New Subscriber: +" + self.__name + " added to the newsletter")
